app/src/main/java/net/programmierecke/radiodroid2/ActivityMain.java

```diff
@@ -594,10 +595,18 @@ public void onFocusChange(View v, boolean hasFocus) {
         menuItemListView.setVisible(false);
         menuItemIconsView.setVisible(false);
         menuItemAddAlarm.setVisible(false);
-        
-        MPDServersRepository mpdServersRepository = new MPDServersRepository(getApplicationContext());
-        LiveData<List<MPDServerData>> mpdServers = mpdServersRepository.getAllServers();
-        menuItemMpd.setVisible(mpdServers.getValue().size() > 0);
+
+        RadioDroidApp radioDroidApp = (RadioDroidApp) getApplication();
+        if (radioDroidApp != null) {
```

Comment:
No need to be this defensive =) if application could not be casted to RadioDroidApp - that's some apocalyptic event. Same goes for MPDClient and repository. All of them must be created at the start of the app.

Reply:
i come from rust-lang, which basically forces you to be defensive all the time :)

Reply:
Ah, yeah, it also have proper syntax for this...

Reply:
I'm just trying to say that in java and here this is pointless and in general could be even harmful, because if something shouldn't be null and you check for null without handling null case - the problem may become hidden instead of failing fast.  

Here nothing will go wrong so it's on your discretion, I'm just voicing my opinion.

But I wish I was working with rust since it force the programmer to take better approaches.

Reply:
i understand, i myself have had problems with finding "silent" bugs. but i still think that non-appearing buttons will keep people using the app while crashes might immediately make them uninstall the app.

Reply:
I agree, but again all our app assumes that this things could not be null, so in this specific case...